src/Clan.ts

```diff
@@ -245,4 +280,67 @@ export class Clan {
     );
     return result.includes("You contributed");
   }
+
+  /**
+   * Take items from the stash
+   * 
+   * This function will also take equivalent foldables if the original item cannot be found
+   * 
+   * @param items Items to take
+   * @returns Items successfully taken
+   */
+  @validate
+  take(items: Item[]): Item[] {
+    return items.filter((item) => {
+      if (have(item)) return false;
+
+      const foldGroup = getFoldGroup(item);
+
+      if (foldGroup.some(fold => have(fold))) {
+        cliExecute(`fold ${item.name}`);
+        return false;
+      }
+
+      refreshStash();
+
+      return [item, ...foldGroup].some((matchingItem) => {
+        if (stashAmount(matchingItem) === 0) return false;
+        if (!takeStash(1, matchingItem)) return false;
+        if (matchingItem !== item) cliExecute(`fold ${item.name}`);
+        return true;
+      });
+    });
+  }
+
+  /**
+   * Put items in the stash
+   * @param items Items to put in the stash
+   * @returns Items successfully put in the stash
+   */
+  @validate
+  put(items: Item[]): Item[] {
+    return items.filter((item) => {
+      retrieveItem(1, item);
+      return putStash(1, item);
+    });
+  }
+
+  /**
+   * Return the monster that is currently in the current clan's fax machine if any
+   */
+  @validate
+  withStash<T>(items: Item[], callback: (borrowedItems: Item[]) => T): T {
+    const borrowed = this.take(items);
+    try {
+      return callback(borrowed);
+    } finally {
+      const returnedItems = this.put(borrowed);
+      const diff = difference(borrowed, returnedItems);
+      if (diff.length > 0) {
+        // eslint-disable-next-line no-unsafe-finally
+        throw new Error(`Failed to return ${diff} to ${this.name}`);
```

Comment:
I don't think we should throw errors in the finally clause. Can we print in a different color instead? That way all items you can't return will be shown.

Reply:
I feel like this is a catastrophic failure so I erred on throwing - I don't actually have a problem with throwing here. I would rather not print from libram, want to leave that to the consumer. What do you think?